homeassistant/components/sighthound/image_processing.py

```diff
@@ -109,17 +123,27 @@ def fire_person_detected_event(self, person):
 
     def save_image(self, image, people, directory):
         """Save a timestamped image with bounding boxes around targets."""
-        img = Image.open(io.BytesIO(bytearray(image))).convert("RGB")
+        try:
+            img = Image.open(io.BytesIO(bytearray(image))).convert("RGB")
+        except UnidentifiedImageError:
```

Comment:
Make sure to test this case in a test. Set a side_effect in the patch.

Reply:
I've added the following but the exception is not being raised..
```python
@pytest.fixture
def mock_bad_image_data():
    """Mock bad image data."""
    with mock.patch(
        "homeassistant.components.sighthound.image_processing.Image.open",
        side_effect=UnidentifiedImageError,
    ) as bad_data:
        yield bad_data
```
Is this because `convert` is also being called at the time?

Reply:
The error is not showing in the logs.. :-/

Reply:
done